homeassistant/components/ecobee/climate.py

```diff
@@ -727,6 +755,63 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the sensors used on a climate for a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
+            msg = f"Invalid climate name, available options are: {', '.join(self.preset_modes)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
```

Comment:
Please use a `ServiceValidationError` instead and it would be nice to add [localization](https://developers.home-assistant.io/docs/internationalization/core/#exceptions) for it.

Please also adopt the other places

Reply:
I will create a separate pull request for localization.